backend/hasher/hasher.go

```diff
@@ -0,0 +1,550 @@
+//go:build !plan9 && !js
+// +build !plan9,!js
+
+// Package hasher implements a checksum handling overlay backend
+package hasher
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/cache"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/fspath"
+	"github.com/rclone/rclone/fs/hash"
+	"go.etcd.io/bbolt"
+)
+
+// Register with Fs
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "hasher",
+		Description: "Hasher will cache hashes",
+		NewFs:       NewFs,
+		CommandHelp: commandHelp,
+		Options: []fs.Option{{
+			Name:     "remote",
+			Required: true,
+			Help:     "Remote to cache hashes for, like myremote:mypath",
+		}, {
+			Name:     "hash_names",
+			Default:  "md5",
+			Advanced: false,
+			Help:     "Space/comma separated list of checksum types to support.",
+		}, {
+			Name:     "auto_size",
```

Comment:
Does this mean that we download files < 1k automatically if they have no checksum?

I think the default should probably be no downloading.

Reply:
ok, will change default to 0